File: backend/app/models/dessert.py
# backend/app/models/dessert.py
from sqlalchemy import Column, Integer, String, Float, Boolean, JSON
from app.db.database import Base

class Dessert(Base):
    __tablename__ = "desserts"

    id = Column(Integer, primary_key=True, index=True)
    name = Column(String, unique=True, nullable=False)
    description = Column(String, default="")
    price = Column(Float, nullable=False)
    category = Column(String, default="other")               # cake, cheesecake, bakery, portion, dessert
    ingredients = Column(JSON, default=[])
    allergens = Column(JSON, default=[])
    available = Column(Boolean, default=True)
    image_url = Column(String, nullable=True)

File: client/lib/models/dessert.dart
class Dessert {
  final int id;
  final String name;
  final String description;
  final double price;
  final String category;
  final List<dynamic> ingredients;
  final List<dynamic> allergens;
  final bool available;
  final String? imageUrl;

  Dessert({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    this.category = 'other',
    this.ingredients = const [],
    this.allergens = const [],
    this.available = true,
    this.imageUrl,
  });

  factory Dessert.fromJson(Map<String, dynamic> json) {
    return Dessert(
      id: json['id'] ?? 0,
      name: json['name'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      category: json['category'] ?? 'other',
      ingredients: json['ingredients'] ?? [],
      allergens: json['allergens'] ?? [],
      available: json['available'] ?? true,
      imageUrl: json['image_url'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'price': price,
      'category': category,
      'ingredients': ingredients,
      'allergens': allergens,
      'available': available,
      'image_url': imageUrl,
    };
  }
}
